src/addon/NUOPC/src/NUOPC_Driver.F90

```diff
@@ -3448,7 +3448,7 @@ recursive subroutine routine_Run(driver, importState, exportState, clock, rc)
   
   !-----------------------------------------------------------------------------
   
-  subroutine routine_executeGridComp(is, i, phase, activeClock, name, userrc, rc)
+  recursive subroutine routine_executeGridComp(is, i, phase, activeClock, name, userrc, rc)
```

Comment:
I needed to add this `recursive` attribute to fix a runtime error when running CTSM. I also added the `recursive` attribute below, for subroutine `routine_executeCplComp`. I'm not sure if that second one was needed. Most subroutines in this module were already declared `recursive`, but please let me know if you think it was wrong for either or both of these to be declared `recursive` (I haven't tried tracing the code to verify that these are really meant to be recursive; my sense is that the recursive calls may arise indirectly via callbacks).

Reply:
No objection. I could easily see how these routines might end up in a recursive call tree. It makes sense for them to carry the `recursive` attribute.